django/db/models/functions/datetime.py

```diff
@@ -64,6 +64,8 @@ def as_sql(self, compiler, connection):
 
     def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize=False, for_save=False):
         copy = super().resolve_expression(query, allow_joins, reuse, summarize, for_save)
+        if isinstance(copy.lhs, ResolvedOuterRef):
+            return copy
         field = copy.lhs.output_field
```

Comment:
Both approaches work but I wonder if we'd want to be a bit more liberal here and simply return `copy` if no `output_field` can be retrieved. 

```suggestion
        field = getattr(copy.lhs, 'output_field', None)
        if field is None:
            return copy
```

It would also avoid having to specify an explicit `output_field` when using a `Func` and `RawSQL` when users usually know what they are doing.

Reply:
For `Func` and `RawSQL` you can pass `output_field`. By the way, I'm for your variant, because in this function only validation which is not critical.

Reply:
I pushed Simon's suggestion and fixed `flake8`/`isort`.